tests/image/image.cpp

```diff
@@ -1887,6 +1887,61 @@ TEST_CASE("wxImage::Paste", "[image][paste]")
 
 }
 
+TEST_CASE("wxImage::RGBtoHSV", "[image][rgbtohsv]")
+{
+    SECTION("RGB(0,0,0) (Black) to HSV")
+    {
+	wxImage::RGBValue rgbBlack = { 0, 0, 0 };
+	wxImage::HSVValue hsvBlack = wxImage::RGBtoHSV(rgbBlack);
+
+	CHECK(hsvBlack.value == Approx(0.0));
+	// saturation and hue are undefined
+    }
+    SECTION("RGB(255,255,255) (White) to HSV")
+    {
+	wxImage::RGBValue rgbWhite = { 255, 255, 255 };
+	wxImage::HSVValue hsvWhite = wxImage::RGBtoHSV(rgbWhite);
+
+	CHECK(hsvWhite.saturation == Approx(0.0));
+	CHECK(hsvWhite.value == Approx(1.0));
+	// hue is undefined
+    }
+    SECTION("RGB(0,255,0) (Green) to HSV")
+    {
+	wxImage::RGBValue rgbGreen = { 0, 255, 0 };
+	wxImage::HSVValue hsvGreen = wxImage::RGBtoHSV(rgbGreen);
+
+	CHECK(hsvGreen.hue == Approx(1.0/3.0));
+	CHECK(hsvGreen.saturation == Approx(1.0));
+	CHECK(hsvGreen.value == Approx(1.0));
+    }
+
+    SECTION("RGB to HSV to RGB")
+    {
+        const int numSamples = 8;
+        wxImage::RGBValue rgbValues[numSamples] = {
+            { 0, 0, 0 },
+            { 10, 10, 10 },
+            { 255, 255, 255 },
+            { 255, 0, 0 },
+            { 0, 255, 0 },
+            { 255, 0, 0 },
```

Comment:
Duplicate entry, presumably should be `0, 0, 255`. BTW, none of these tests fail for me on Linux x64 even without the fix.

Reply:
Yes, sorry for being unclear... by "trivial" I still meant something showing what doesn't work now and fixing it. Tests for the cases already working correctly are welcome too, of course, but it would be even more useful to have tests for the currently broken cases.

Also, could you please fix indentation/style (put braces on their own lines)? TIA!

Reply:
{1, 2, 3} is a testcase failing for x87, as I wrote. This is a testcase from wxPython, currently failing.

Reply:
I'm sorry but I just don't see it. I've built your tests _without_ applying the fix using MSVS 2019 in 32 bit build and I don't see any failures. What exactly do I need to do to reproduce them?

P.S. I'm force pushing the fixes to indentation and C++11-isms here.

Reply:
I see the issue with various GCC versions on i568.
Maybe you use `--fast-math` or anything else which forces it to use SSE?